matrix_sdk/src/client.rs

```diff
@@ -429,6 +430,22 @@ impl Client {
         session.as_ref().cloned().map(|s| s.user_id)
     }
 
+    /// Fetches the display name of the owner of the client.
+    pub async fn user_display_name(&self) -> Result<Option<String>> {
```

Comment:
I think we can leave out the `user` part from the method name here since it's implied that we're going to do this for our own user if no user is specified.

Also we'll have a method to get the display name of another user as well, in that case we're gonna have problems figuring out a name for that method.

```suggestion
    pub async fn display_name(&self) -> Result<Option<String>> {
```